toto/verifylib.py

```
@@ -213,11 +213,15 @@ def verify_command_alignment(command, expected_command):
 
   for i in range(min(command_len, expected_command_len)):
     if command[i] != expected_command[i]:
-      raise Exception("Command '%s' and expected command '%s' do not align" \
-          % (command, expected_command))
+      raise CommandAlignmentFailed(
+          "Command '{0}' and expected command '{1}' do not align"
+          .format(command, expected_command))
   if command_len != expected_command_len:
-    log.warning("Command '%s' and expected command '%s' do not fully align" \
-        % (command, expected_command))
+    # FIXME we don't want print statements in a library
+    # The warning will be obsolete with a fix for
+    # https://github.com/in-toto/in-toto/issues/46
+    print("Command '{0}' and expected command '{1}' do not fully align"
```

I'm curious, why don't we use the logger instance as it was done before? 
`log.warning` sounded like a good idea.

---

Hm. I talked to @vladimir-v-diaz about it. We agreed that the `logging` module should not be used to give direct feedback to the user, but rather to *log* errors/warnings/infos to some place (could be a file) intended for the developer/debugger.

---

We figured that if in-toto is going to be used as a library, the calling application probably doesn't want to see messages from the library printed/logged to the console (at least not by default).  For applications that would like to have these log statements show up on the console, perhaps an option can be added for them?

---

Ok, then it would make sense to add exceptions to pr #40 and let the user-facing tool handle them?

---

So I re-thought all of this and summarized my conclusions in https://github.com/in-toto/in-toto/issues/6#issuecomment-263988714. Nevertheless, this still does not solve this particular issue. If we raise an exception instead of a warning here, the exception could be caught and translated to a warning in the user tool, but this would also abort the the layout verification, which we don't want.

---

Solutions:
 - return a list of "pending command verification exceptions"  that can be raised, caught and logged once the entire verification has finished
 - just log the warning in the library